python/pyspark/sql/context.py

```diff
@@ -315,28 +335,19 @@ def createDataFrame(self, data, schema=None, samplingRatio=None):
         else:
             rdd = data
 
-        if schema is None:
-            schema = self._inferSchema(rdd, samplingRatio)
+        if not isinstance(schema, StructType):
```

Comment:
i think we should only do this if schema is None. 

and if schema is not None, and is not a StructType, we should throw a TypeError